pkg/kubelet/kubelet.go

```diff
@@ -2678,12 +2678,20 @@ func getPodReadyCondition(spec *api.PodSpec, containerStatuses []api.ContainerSt
 		unreadyMessages = append(unreadyMessages, fmt.Sprintf("containers with unknown status: %s", unknownContainers))
 	}
```

Comment:
If you accept my suggestions below to not print unreadyMessages, then you move the `if podPhase == api.PodSucceeded` checks to here, avoiding duplication of that check.